src/rapids_dependency_file_generator/rapids_dependency_file_generator.py

```diff
@@ -35,9 +35,12 @@ def delete_existing_files(root="."):
         for fn in filter(
             lambda fn: fn.endswith(".txt") or fn.endswith(".yaml"), filenames
         ):
-            with open(file_path := os.path.join(dirpath, fn)) as f:
-                if HEADER in f.read():
-                    os.remove(file_path)
+            try:
+                with open(file_path := os.path.join(dirpath, fn)) as f:
+                    if HEADER in f.read():
+                        os.remove(file_path)
+            except UnicodeDecodeError:
+                pass
```

Comment:
I don't think we want to skip cleaning the file in this case. I'm not sure what we want, though. If you can provide an example file, I would like to dig into this and try it locally before approving this approach.